webui/react/src/pages/ExperimentDetails/ExperimentSingleTrialTabs.tsx

```diff
@@ -77,12 +78,16 @@ const ExperimentSingleTrialTabs: React.FC<Props> = (
 
   const fetchFirstTrialId = useCallback(async () => {
     try {
+      // make sure the experiment is in terminal state before the request is made.
+      const isTerminalExp = terminalRunStates.has(experiment.state);
```

Comment:
question: if `isTerminalExp` is true, why not just immediately set `wontHaveTrials` to true and return?

Reply:
to make sure we at least check once. it could be that the experiment is in a terminal state but has trials.